File: typescript/base-hmi-types/src/screens/shapes/HmiEllipticalArc.ts
import { HmiProperty, staticProperty } from "../base.js";
import { HmiEllipticalShapeBase } from "./HmiEllipticalShapeBase.js";
import { HmiObjectType } from "../base/HmiObjectType.js";

export class HmiEllipticalArc extends HmiEllipticalShapeBase {
  constructor() {
    super();
    this.hmiObjectType = HmiObjectType.HmiEllipticalArc;
  }

  startAngle: HmiProperty<number> = staticProperty(0);
  sweepAngle: HmiProperty<number> = staticProperty(0);
}

File: src/BaseHmiTypes/Screens/Shapes/HmiEllipticalArc.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Shapes;

public class HmiEllipticalArc : HmiEllipticalShapeBase
{
    public HmiEllipticalArc()
    {
        HmiObjectType = BaseHmiTypes.Screens.Base.HmiObjectType.HmiEllipticalArc;
    }

    public HmiProperty<double> StartAngle { get; set; } = 0;

    public HmiProperty<double> SweepAngle { get; set; } = 0;
}
